minefield_game/cell.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+enum class CellStatus : char
+{
+    Empty = 'o',   // Empty cell
+    Guess = '?',   // A guess has been made
+    Bomb  = '*',   // Contains a mine
+    Taken = 'x'    // Disabled cell (can no longer be used)
+};
+
+
+struct Cell
+{
+
+    int x;
+    int y;
+    CellStatus status;
+
+    void setStatus(CellStatus newStatus);
+    CellStatus getStatus() const;
+    bool isCellTaken() const;
```

Comment:
#should make it a free function - the method does not access or modify any `private` member

```cpp
bool isCellTaken(Cell const& cell)
{
    return cell.status == CellStatus::Taken;
}
```